### PR TITLE
Added Battery Widget

### PR DESCRIPTION
**Pull request by vkylamba**
Updated.
image

### CODE REVIEW

**Comment by Paul-Reed** [comment]
Thanks for the PR, however it doesn't seem to work here as intended. For example; The widget is configured like this;

settings

and the feed input is 151, I get;

battery

Also, I see that the text size does not scale with the size of the widget, or is user definable - font, label colour (see the bar widget - also in the screenshot above)
It would be good to have some formatting consistency across the widgets.

Paul

**Comment by vkylamba** [comment]
@Paul-Reed Thanks for the review. I have updated the PR. Please have a look.

**Comment by Paul-Reed** [comment]
That's better, it displays the animation OK now.

1. Do we need a widget title? we could of course have several battery widgets displayed on the same page (such as banks of batteries) but with no title, it may not be easy to identify which widget relates to which battery.
2. I think variable rounding of the displayed value would be useful, such as https://github.com/emoncms/dashboard/blob/master/widget/bar/bar_render.js#L252-L258 to enable different precision depending upon the value size. If you were monitoring 999V is there any need to display precision to 0.1V. Similarly for a 1V battery it would be useful to display the value to greater precision.
3. The text value does not appear to be centre aligned to the widget - see screenshot below using Arial Back typeface.
4. Also, for consistency, could the font size be increased to match that of the bar widget? In the screenshot, both widgets are the same size, yet the battery text size is smaller. (so that we have some dashboard consistency)

Paul

align

**Comment by vkylamba** [comment]
1. Added widget title. 
2. Corrected the rounding off. 
3. Text centre aligned.
4. Fixed font size.
image

File: widget/battery/battery_render.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+  var fontoptions = [
+    [9, "Arial Black"],
+    [8, "arial"],
+    [7, "Arial Narrow"],
+    [6, "sans-serif"],
+    [5, "Helvetica"],
+    [4, "Comic Sans MS"],
+    [3, "Courier New"],
+    [2, "arial"],
```
Comment (Paul-Reed) [inline widget/battery/battery_render.js]: arial is already listed on line 40

File: widget/battery/battery_render.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+      if (font == 0){fontname = "Impact"}
+      if (font == 1){fontname = "Georgia"}
+      if (font == 2){fontname = "arial"}
+      if (font == 3){fontname = "Courier New"}
+      if (font == 4){fontname = "Comic Sans MS"}
+      if (font == 5){fontname = "Helvetica"}
+      if (font == 6){fontname = "sans-serif"}
+      if (font == 7){fontname = "Arial Narrow"}
+      if (font == 8){fontname = "Arial"}
```
Comment (Paul-Reed) [inline widget/battery/battery_render.js]: Arial is already listed on line 95

**Comment by Paul-Reed** [comment]
Thanks for your patience Vikas, the battery widget looks good and matches the existing widgets nicely.

align

Paul

**Comment by Paul-Reed** [comment]
Vikas, I've made a forum entry about the widget https://community.openenergymonitor.org/t/battery-widget-now-added-to-dashboard/3473
I can't find your ID in the forum, are you a forum member? It would be good to introduce yourself in the thread.

Thanks

Paul

**Comment by vkylamba** [comment]
Thank you Paul for merging the PR. I wasn't a member till now. Just signed up.

**Comment by glynhudson** [comment]
Nice work, looks good. Thanks guys!

- sent from my mobile device

**Pull request merged**